### PR TITLE
PLF-8460 : Problem in eXo push notification addon : cookie array is null

### PR DESCRIPTION
**Pull request by rdenarie**
This change is needed because an NPE can be thrown in getCookieValue if cookie array is null
This change add a test before using the array to check if it is null or not

### CODE REVIEW

File: service/src/test/java/org/exoplatform/push/filter/AuthenticationHeaderFilterTest.java [hunk 1]
```diff
@@ -117,4 +117,18 @@ public void shouldReturnConcatenatedHeaderWithUserAuthenticatedAndMissingCookies
     assertTrue(authenticationHeaderTokens.contains("JSESSIONID=valueJSESSIONID"));
     assertTrue(authenticationHeaderTokens.contains("JSESSIONIDSSO=valueJSESSIONIDSSO"));
   }
+
+  @Test
+  public void shouldNotThrowsNPE() {
```
Comment (thomasdelhomenie) [inline service/src/test/java/org/exoplatform/push/filter/AuthenticationHeaderFilterTest.java]:
```suggestion
  public void shouldNotThrowsNPEWhenGettingCookieValueWithNullArray() {
```

File: service/src/test/java/org/exoplatform/push/filter/AuthenticationHeaderFilterTest.java [hunk 2]
```diff
@@ -117,4 +117,18 @@ public void shouldReturnConcatenatedHeaderWithUserAuthenticatedAndMissingCookies
...
+
+    AuthenticationHeaderFilter filter = new AuthenticationHeaderFilter();
+    try {
+      filter.getCookieValue(null,"test");
+      assertTrue(Boolean.TRUE);
```
Comment (thomasdelhomenie) [inline service/src/test/java/org/exoplatform/push/filter/AuthenticationHeaderFilterTest.java]: This assertion is useless. I would test the value returned by the previous line instead (should be null).

File: service/src/test/java/org/exoplatform/push/filter/AuthenticationHeaderFilterTest.java [hunk 3]
```diff
@@ -117,4 +117,18 @@ public void shouldReturnConcatenatedHeaderWithUserAuthenticatedAndMissingCookies
     assertTrue(authenticationHeaderTokens.contains("JSESSIONID=valueJSESSIONID"));
     assertTrue(authenticationHeaderTokens.contains("JSESSIONIDSSO=valueJSESSIONIDSSO"));
   }
+
+  @Test
+  public void shouldNotThrowsNPE() {
+
+    AuthenticationHeaderFilter filter = new AuthenticationHeaderFilter();
+    try {
+      filter.getCookieValue(null,"test");
```
Comment (thomasdelhomenie) [inline service/src/test/java/org/exoplatform/push/filter/AuthenticationHeaderFilterTest.java]: We generally do not test protected/private methods. Instead we should test the doFilter method with a mocked request returning a null cookies array IMO.

Comment (rdenarie) [reply service/src/test/java/org/exoplatform/push/filter/AuthenticationHeaderFilterTest.java]: Done

File: service/src/test/java/org/exoplatform/push/filter/AuthenticationHeaderFilterTest.java (hunk 2 above)
Comment (rdenarie) [reply service/src/test/java/org/exoplatform/push/filter/AuthenticationHeaderFilterTest.java]: With the last change, I could only test with assert(true), I cannot test the return of dofilter

Comment (rdenarie) [reply service/src/test/java/org/exoplatform/push/filter/AuthenticationHeaderFilterTest.java]: Done
